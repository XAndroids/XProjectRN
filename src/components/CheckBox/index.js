import React from 'react'
import {TouchableOpacity, Image} from 'react-native'

import PropTypes from "prop-types";

//函数式组件:XCheckBox
//参考：https://github.com/XLibrarySources/react-native-elements/blob/next/src/checkbox/CheckBox.js
const CheckBox = (props) => {
    const icon = props.isChecked ? props.checkedIcon : props.uncheckedIcon;

    return (
        <TouchableOpacity onPress={props.onCheckedChange}>
            <Image testID={'checkbox_image'} style={[props.style]} source={icon}/>
        </TouchableOpacity>
    );
};

CheckBox.propTypes = {
    isChecked: PropTypes.bool.isRequired,//没有默认值，都声明呈required
    checkedIcon: PropTypes.object.isRequired,//FIXME 这是用PropTypes.object检测是否合适？？
    uncheckedIcon: PropTypes.object.isRequired,
    onCheckedChange: PropTypes.func
};

export default CheckBox
