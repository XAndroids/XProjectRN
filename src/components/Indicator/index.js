import React from 'react'
import {View} from 'react-native'
import PropTypes from 'prop-types'
import {styles} from "./styles"

const Indicator = (props) => {
    let circles = [];
    let {currentIndex, circleCount} = props;

    for (let i = 0; i < circleCount; i++) {
        let style = styles.circle;
        if (i === currentIndex) {
            style = [styles.circle, styles.circleSelected];
        }
        circles.push(<View testID={'indicator_view'} style={style} key={i}/>);
    }

    return (
        <View style={styles.circleContainer}>
            {circles}
        </View>
    )
};

Indicator.propTypes = {
    circleCount: PropTypes.number,
    currentIndex: PropTypes.number,
};

export default Indicator
