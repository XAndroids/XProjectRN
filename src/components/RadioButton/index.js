import React from 'react'
import {TouchableOpacity, View} from 'react-native'
import PropTypes from 'prop-types'
import {styles} from './styles'

const DEFAULT_SIZE_MULTIPLIER = 0.7;
const DEFAULT_OUTER_BORDER_WIDTH_MULTIPLIER = 0.15;
/**
 * 单选按钮
 * 参考：https://github.com/mmazzarolo/react-native-radio-button/blob/master/RadioButton.js
 */
const RadioButton = (props) => {
    const {size, innerColor, outerColor, isSelected, onPress} = props;

    //外圈样式
    const outerStyle = {
        borderColor: outerColor,
        width: size + size * DEFAULT_SIZE_MULTIPLIER,
        height: size + size * DEFAULT_SIZE_MULTIPLIER,
        borderRadius: (size + size * DEFAULT_SIZE_MULTIPLIER) / 2,
        borderWidth: size * DEFAULT_OUTER_BORDER_WIDTH_MULTIPLIER
    };

    //内圈样式
    const innerStyle = {
        width: size,
        height: size,
        borderRadius: size / 2,
        backgroundColor: innerColor
    };

    return (
        <TouchableOpacity testID="radiobutton_touchableopacity_outer" style={[props.style, styles.radio, outerStyle]}
                          onPress={onPress}>
            {isSelected ? <View testID="radiobutton_view_inner" style={innerStyle}/> : null}
        </TouchableOpacity>
    )
};

RadioButton.propTypes = {
    size: PropTypes.number,
    innerColor: PropTypes.string,
    outerColor: PropTypes.string,
    isSelected: PropTypes.bool,
    onPress: PropTypes.func
};

RadioButton.defaultProps = {
    size: 16,
    innerColor: '#058b7d',
    outerColor: '#058b7d',
    isSelected: false,
    onPress: () => null
};

export default RadioButton

