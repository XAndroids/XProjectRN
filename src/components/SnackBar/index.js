import {NativeModules, processColor} from 'react-native';


const SnackBar = {
    //常量
    LENGTH_LONG: NativeModules.RNSnackbar.LENGTH_LONG,
    LENGTH_SHORT: NativeModules.RNSnackbar.LENGTH_SHORT,
    LENGTH_INDEFINITE: NativeModules.RNSnackbar.LENGTH_INDEFINITE,
    /**
     * 显示SnackBar
     */
    show(options) {
        const text = options.text;
        const textColorRaw = options.textColor;
        const textColor = textColorRaw && processColor(textColorRaw);
        const backgroundColor = options.backgroundColor && processColor(options.backgroundColor);

        const action = options.action || {};
        const actionText = action.text;
        const actionTextColorRaw = action.textColor;
        const actionTextColor = actionTextColorRaw && processColor(actionTextColorRaw);

        const onPressCallback = action.onPress || (() => {
        });

        const nativeOptions = {
            ...options,
            text,
            textColor,
            backgroundColor,
            action: options.action ? {
                ...action,
                text: actionText,
                textColor: actionTextColor,
            } : undefined,
        };
        console.log(JSON.stringify(NativeModules));
        NativeModules.RNSnackbar.show(nativeOptions, onPressCallback);
    },

    /**
     * 隐藏SnackBar
     */
    dismiss() {
        NativeModules.RNSnackbar.dismiss();
    },
};

export default SnackBar;
