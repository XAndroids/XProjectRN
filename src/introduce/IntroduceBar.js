import React, {Component} from 'react'
import {View, Text, Image, TouchableOpacity, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';
import XIndicator from '../component/XIndicator';
import {getWindowsWidth} from '../utils/Devices';

/**
 * 介绍页面底部栏，负责ViewPager指示器和导航按钮
 */
export default class IntroduceBar extends Component {

    constructor(props) {
        super(props);

        this.onSkipPress = this.onSkipPress.bind(this);
        this.onNextPress = this.onNextPress.bind(this);
        this.onDonePress = this.onDonePress.bind(this);
    }

    render() {
        return (
            <View style={[this.props.style, {
                flexDirection: 'row',
                alignItems: 'center',
                height: 50,
                width: getWindowsWidth()
            }]}>
                {this.renderLeftButton()}
                <XIndicator circleCount={this.props.showPageCount} currentIndex={this.props.showPageIndex}/>
                {this.renderRightButton()}
            </View>
        )
    }

    renderLeftButton() {
        if (this.props.showPageIndex === 0 || this.props.showPageIndex === 1) {
            return (
                <TouchableOpacity style={styles.touchable} tonPress={this.onSkipPress}>
                    <Text style={[styles.text, {alignSelf: 'flex-start', marginLeft: 30}]}>SKIP</Text>
                </TouchableOpacity>
            );
        } else if (this.props.showPageIndex === 2) {
            return (
                <View style={{flex: 1}}/>
            );
        }
    }

    renderRightButton() {
        if (this.props.showPageIndex === 0 || this.props.showPageIndex === 1) {
            return (
                <TouchableOpacity style={styles.touchable} onPress={this.onNextPress}>
                    <Image style={{alignSelf: 'flex-end', width: 50, height: 50}}
                           source={require('../../img/introduce_nextbutton_background.png')}/>
                </TouchableOpacity>
            );
        } else if (this.props.showPageIndex === 2) {
            return (
                <TouchableOpacity style={styles.touchable} onPress={this.onDonePress}>
                    <Text style={[styles.text, {alignSelf: 'flex-end', marginRight: 30}]}>DONE</Text>
                </ TouchableOpacity>
            );
        }

    }

    onSkipPress() {
        this.props.navigation.navigate('Main');
    }

    onNextPress() {
        this.props.onNextPress(this.props.showPageIndex + 1);
    }

    onDonePress() {
        this.props.navigation.navigate('Main');
    }
}

IntroduceBar.propTypes = {
    showPageIndex: PropTypes.number,
    showPageCount: PropTypes.number,
    onNextPress: PropTypes.func
};

IntroduceBar.defaultProps = {
    showPageIndex: 0,
    showPageCount: 0,
};

const styles = StyleSheet.create({
    text: {
        color: 'white'
    },
    touchable: {
        flex: 1
    }
});
