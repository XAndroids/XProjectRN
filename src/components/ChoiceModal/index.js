import React, {Component} from 'react'
import {FlatList, Modal, Text, View, TouchableWithoutFeedback} from 'react-native'
import {RadioButton} from '../../components'
import PropTypes from 'prop-types'

/**
 * 单选Modal
 */
export default class ChoiceModal extends Component {
    constructor(props) {
        super(props);
        this._onTypeItemPress = this._onTypeItemPress.bind(this)
    }

    render() {
        const backgroundStyle = {
            backgroundColor: this.props.visible ? 'rgba(0, 0, 0, 0.5)' : '#f5fcff'
        };

        return (
            <Modal testID={this.props.testID}
                   visible={this.props.visible}
                   transparent={true}
                   onRequestClose={this.props.onRequestClose}>
                <View style={[{
                    flex: 1,
                    justifyContent: 'center',
                    paddingLeft: 25,
                    paddingRight: 25,
                    paddingTop: 42,
                    paddingBottom: 42
                }, backgroundStyle]}>
                    <View style={[{backgroundColor: 'white'}]}>
                        <Text style={{
                            height: 45,
                            color: 'black',
                            fontSize: 20,
                            fontWeight: 'bold',
                            textAlignVertical: 'bottom',
                            marginLeft: 30,
                            marginBottom: 8
                        }}>Select Publisher</Text>
                        <FlatList data={this.props.choiceList}
                                  renderItem={this._renderItemComponent}
                                  keyExtractor={(item, index) => index.toString()}/>
                    </View>
                </View>
            </Modal>
        );
    }

    _renderItemComponent = ({item, index}) => {
        return (
            <TouchableWithoutFeedback onPress={() => this._onTypeItemPress(item, index)}>
                <View style={{flexDirection: 'row', height: 47}}>
                    <RadioButton style={{marginLeft: 23, marginRight: 23}} isSelected={this.props.choiceIndex === index}
                                 size={12}/>
                    <Text style={{
                        flex: 1,
                        color: 'black',
                        fontSize: 18,
                        textAlignVertical: 'center'
                    }}>{item.name}</Text>
                </View>
            </TouchableWithoutFeedback>
        );
    };

    _onTypeItemPress(item, index) {
        this.props.onSelectedTypeChanged(item, index);
    }
}

ChoiceModal.defaultProps = {
    visible: false,
    choiceIndex: -1,
};

ChoiceModal.propTypes = {
    testID: PropTypes.string,
    visible: PropTypes.bool,
    choiceList: PropTypes.array.isRequired,
    choiceIndex: PropTypes.number,
    onRequestClose: PropTypes.func,
    onSelectedTypeChanged: PropTypes.func
};
