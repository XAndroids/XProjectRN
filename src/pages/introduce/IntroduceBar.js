import React, {Component} from 'react'
import {View, Text, Image, TouchableOpacity, StyleSheet} from 'react-native'
import PropTypes from 'prop-types'
import {Indicator} from '../../components'
import {getWindowsWidth} from '../../utils'

/**
 * 介绍页面底部栏，负责ViewPager指示器和导航按钮
 */
export default class IntroduceBar extends Component {

    constructor(props) {
        super(props);

        this._onSkipPress = this._onSkipPress.bind(this);
        this._onNextPress = this._onNextPress.bind(this);
        this._onDonePress = this._onDonePress.bind(this);
    }

    render() {
        const {showPageCount, showPageIndex, style} = this.props;

        return (
            <View style={[style, {
                flexDirection: 'row',
                alignItems: 'center',
                height: 50,
                width: getWindowsWidth()
            }]}>
                {this._renderLeftButton()}
                <Indicator circleCount={showPageCount} currentIndex={showPageIndex}/>
                {this._renderRightButton()}
            </View>
        )
    }

    _renderLeftButton() {
        const {showPageIndex} = this.props;

        if (showPageIndex === 0 || showPageIndex === 1) {
            return (
                <TouchableOpacity testID={'introduce_button_skip'} style={styles.touchable} onPress={this._onSkipPress}>
                    <Text testID='skip' style={[styles.text, {alignSelf: 'flex-start', marginLeft: 30}]}>SKIP</Text>
                </TouchableOpacity>
            );
        } else if (showPageIndex === 2) {
            return (
                <View style={{flex: 1}}/>
            );
        }
    }

    _renderRightButton() {
        const {showPageIndex} = this.props;

        if (showPageIndex === 0 || showPageIndex === 1) {
            return (
                <TouchableOpacity testID={'introduce_button_next'} style={styles.touchable} onPress={this._onNextPress}>
                    <Image style={{alignSelf: 'flex-end', width: 50, height: 50}}
                           source={require('../../../img/introduce_nextbutton_background.png')}/>
                </TouchableOpacity>
            );
        } else if (showPageIndex === 2) {
            return (
                <TouchableOpacity style={styles.touchable} onPress={this._onDonePress}>
                    <Text style={[styles.text, {alignSelf: 'flex-end', marginRight: 30}]}>DONE</Text>
                </TouchableOpacity>
            );
        }

    }

    _onSkipPress() {
        this.props.navigation.navigate('Main');
    }

    _onNextPress() {
        this.props.onNextPress(this.props.showPageIndex + 1);
    }

    _onDonePress() {
        this.props.navigation.navigate('Main');
    }
}

IntroduceBar.propTypes = {
    showPageIndex: PropTypes.number,
    showPageCount: PropTypes.number,
    onNextPress: PropTypes.func
};

IntroduceBar.defaultProps = {
    showPageIndex: 0,
    showPageCount: 0,
};

const styles = StyleSheet.create({
    text: {
        color: 'white'
    },
    touchable: {
        flex: 1
    }
});
