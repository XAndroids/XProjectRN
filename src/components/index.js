import CheckBox from './CheckBox'
import Indicator from './Indicator'
import RadioButton from './RadioButton'
import Toolbar from './Toolbar'
import TypeChoiceModal from './TypeChoiceModal'
import ViewPager from './ViewPager'
import CardView from './CardView'
import SnackBar from "./SnackBar";

export {
    CheckBox,
    Indicator,
    RadioButton,
    Toolbar,
    TypeChoiceModal,
    ViewPager,
    CardView,
    SnackBar
}
