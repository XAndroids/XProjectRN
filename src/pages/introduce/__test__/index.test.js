import React from "react";
import {View} from "react-native"
import {render, shallow} from "react-native-testing-library";
import Introduce from "../index";
import {ViewPager} from "../../../components";
import IntroduceBar from "../IntroduceBar";

describe("introduce page test",()=>{
    it("introduce page show test", () => {
        const {getByType} = render(<Introduce/>);

        //验证rootView是否显示正常
        const rootView= getByType(View)
        expect(rootView).not.toBeNull()
        expect(rootView.props.style).toEqual({ flex: 1 })

        //验证ViewPager是否显示正常
        const viewPager = getByType(ViewPager)
        expect(viewPager.props.initPageIndex).toBe(0)
        expect(viewPager.props.children.length).toBe(3)
        expect(viewPager.props.children[0].props).toEqual( {
            title: 'Welcome',
            image: { testUri: '../../../img/introduce_icon1.png' },
            content: 'Thank you for installing Paperboy.What makes Paperboy different form other news reader apps is its simplicity and elegant design.',
            backgroundColor: '#01A3AE'
        })
        expect(viewPager.props.children[1].props).toEqual( {
                title: 'Customizations',
                image: { testUri: '../../../img/introduce_icon2.png' },
                content: 'We believe that each user is unique and hence several customization options are provided to suit different reading styles.To customize please visit the settings page.',
                backgroundColor: '#4CAF50'
            }
        )
        expect(viewPager.props.children[2].props).toEqual( {
                title: 'Reading pattern learner',
                image: { testUri: '../../../img/introduce_icon3.png' },
                content: 'The home screen tiles will automatically reaarrange based on reading patterns for better user experence.All data is stored locally kepping in mind user privacy.',
                backgroundColor: '#673AB8'
            }
        )

        //验证IntroduceBar是否显示正常
        const introduceBar = getByType(IntroduceBar)
        expect(introduceBar).not.toBeNull()
        expect(introduceBar.props.style).toEqual({ position: 'absolute', bottom: 0 })
        expect(introduceBar.props.showPageIndex).toBe(0)
        expect(introduceBar.props.showPageCount).toBe(3)
    })

    describe("introduce page Snapshot test",()=>{
        const {toJSON} = render(<Introduce/>);
        expect(toJSON()).toMatchSnapshot()
    })

    describe("introduce page Shallow test",()=>{
        const {output} = shallow(<Introduce/>)
        expect(output).toMatchSnapshot()
    })
})
