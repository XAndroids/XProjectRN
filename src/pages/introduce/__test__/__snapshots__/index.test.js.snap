// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    horizontal={true}
    onMomentumScrollEnd={[Function]}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "#01A3AE",
              "height": 1334,
              "width": 750,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 28,
                },
              ]
            }
          >
            Welcome
          </Text>
        </View>
        <Image
          source={
            Object {
              "testUri": "../../../img/introduce_icon1.png",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 16,
                },
              ]
            }
          >
            Thank you for installing Paperboy.What makes Paperboy different form other news reader apps is its simplicity and elegant design.
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "height": 1,
              "marginBottom": 50,
              "marginTop": 20,
              "width": 750,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "#4CAF50",
              "height": 1334,
              "width": 750,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 28,
                },
              ]
            }
          >
            Customizations
          </Text>
        </View>
        <Image
          source={
            Object {
              "testUri": "../../../img/introduce_icon2.png",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 16,
                },
              ]
            }
          >
            We believe that each user is unique and hence several customization options are provided to suit different reading styles.To customize please visit the settings page.
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "height": 1,
              "marginBottom": 50,
              "marginTop": 20,
              "width": 750,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "#673AB8",
              "height": 1334,
              "width": 750,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 28,
                },
              ]
            }
          >
            Reading pattern learner
          </Text>
        </View>
        <Image
          source={
            Object {
              "testUri": "../../../img/introduce_icon3.png",
            }
          }
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "white",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 16,
                },
              ]
            }
          >
            The home screen tiles will automatically reaarrange based on reading patterns for better user experence.All data is stored locally kepping in mind user privacy.
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.1)",
              "height": 1,
              "marginBottom": 50,
              "marginTop": 20,
              "width": 750,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "position": "absolute",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 50,
          "width": 750,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID="introduce_button_skip"
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
            },
            Object {
              "alignSelf": "flex-start",
              "marginLeft": 30,
            },
          ]
        }
        testID="skip"
      >
        SKIP
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "height": 10,
              "marginHorizontal": 5,
              "width": 10,
            },
            Object {
              "backgroundColor": "#0f0",
            },
          ]
        }
        testID="indicator_view"
      />
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 5,
            "width": 10,
          }
        }
        testID="indicator_view"
      />
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "height": 10,
            "marginHorizontal": 5,
            "width": 10,
          }
        }
        testID="indicator_view"
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flex": 1,
          "opacity": 1,
        }
      }
      testID="introduce_button_next"
    >
      <Image
        source={
          Object {
            "testUri": "../../../img/introduce_nextbutton_background.png",
          }
        }
        style={
          Object {
            "alignSelf": "flex-end",
            "height": 50,
            "width": 50,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[` 2`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewPager
    initPageIndex={0}
    onPageSwitch={[Function]}
    viewPagerRef={[Function]}
  >
    <IntroducePage
      backgroundColor="#01A3AE"
      content="Thank you for installing Paperboy.What makes Paperboy different form other news reader apps is its simplicity and elegant design."
      image={
        Object {
          "testUri": "../../../img/introduce_icon1.png",
        }
      }
      title="Welcome"
    />
    <IntroducePage
      backgroundColor="#4CAF50"
      content="We believe that each user is unique and hence several customization options are provided to suit different reading styles.To customize please visit the settings page."
      image={
        Object {
          "testUri": "../../../img/introduce_icon2.png",
        }
      }
      title="Customizations"
    />
    <IntroducePage
      backgroundColor="#673AB8"
      content="The home screen tiles will automatically reaarrange based on reading patterns for better user experence.All data is stored locally kepping in mind user privacy."
      image={
        Object {
          "testUri": "../../../img/introduce_icon3.png",
        }
      }
      title="Reading pattern learner"
    />
  </ViewPager>
  <IntroduceBar
    onNextPress={[Function]}
    showPageCount={3}
    showPageIndex={0}
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
      }
    }
  />
</Component>
`;
